provider/datagen/src/driver.rs

```diff
@@ -847,17 +849,21 @@ fn select_locales_for_marker(
                         );
                         selected_langids.insert(parent_langid);
                     }
-                    if let Some(parent_locales) = maybe_parent_locales {
-                        for morphed_req in parent_locales.iter() {
+                    if let Some(parent_ids) = maybe_parent_ids {
+                        for morphed_id in parent_ids.iter() {
                             // Special case: don't pull extensions or aux keys up from the root.
-                            if morphed_req.0.is_langid_und()
-                                && !(morphed_req.0.is_empty() && morphed_req.1.is_empty())
+                            if morphed_id.locale.is_langid_und()
+                                && !(morphed_id.locale.is_empty()
+                                    && morphed_id.marker_attributes.is_empty())
```

Comment:
Suggestion: can you write this as `!morphed_id.is_empty()` ?

Reply:
I don't really want to define too much logic on `DataIdentifier`